test/receiver.spec.ts

```diff
@@ -40,6 +40,61 @@ describe("Receiver", () => {
     assert.isFalse(receiver.isOpen(), "Receiver should not be open.");
   });
 
+  it(".remove() removes event listeners", async () => {
+    const receiver = await connection.createReceiver();
+    receiver.on(rhea.ReceiverEvents.receiverOpen, () => {
+      /** no-op */
+    });
+
+    assert.isTrue(receiver.isOpen(), "Receiver should be open.");
+    assert.isAtLeast(receiver.listenerCount(rhea.ReceiverEvents.receiverOpen), 1);
+
+    receiver.remove();
+
+    // TODO: receiver.isOpen() returns true at this point, need to find out why
+    // assert.isFalse(receiver.isOpen(), "Receiver should not be open.");
+    assert.strictEqual(receiver.listenerCount(rhea.ReceiverEvents.receiverOpen), 0);
+
+  });
+
+  it(".close() removes event listeners", async () => {
+    const receiver = await connection.createReceiver();
+    receiver.on(rhea.ReceiverEvents.receiverOpen, () => {
+      /** no-op */
+    });
+
+    assert.isAtLeast(receiver.listenerCount(rhea.ReceiverEvents.receiverOpen), 1);
+
+    await receiver.close();
+
+    assert.strictEqual(receiver.listenerCount(rhea.ReceiverEvents.receiverOpen), 0);
+  });
+
+  // TODO: This test fails because we first get a receiver_open event instead of
```

Comment:
I'm not sure I understand what's happening. You're seeing the client receiver_open get fired after the service sender_open is fired?

Reply:
- service fires sender_open, in the call back, the sender is close with an error
- client fires receiver_open. rhea-promise resolves the promise, removes all listeners
- client fires receiver_close, gets the error that was sent by the service, but rhea-promise has no listeners for this anymore



In the same test for sender:
- service fires receiver_open, in the call back, the receiver is close with an error
- client fires sender_close, gets the error that was sent by the service

Reply:
Ah I see. In this case then we aren't seeing a difference in behaviour after your change right? I was confused at first this had something to do with how you changed when removeListeners is called but this scenario doesn't sound like it'd be impacted by your change.

Reply:
Ya, no diff in behavior after my change.
My change was in the code path where the client explicitly closes the link.
This is a case where service closes the link with an error